commit 8286274284e15b11b0f531b6ceeef21fbe00a8dd
Author: Herbert Xu <herbert@gondor.apana.org.au>
Date:   Mon May 24 00:14:10 2010 -0700

    tun: Update classid on packet injection
    
    This patch makes tun update its socket classid every time we
    inject a packet into the network stack.  This is so that any
    updates made by the admin to the process writing packets to
    tun is effected.
    
    Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/net/core/sock.c b/net/core/sock.c
index a05ae7f9771e..37fe9b6adade 100644
--- a/net/core/sock.c
+++ b/net/core/sock.c
@@ -1064,6 +1064,7 @@ void sock_update_classid(struct sock *sk)
 	if (classid && classid != sk->sk_classid)
 		sk->sk_classid = classid;
 }
+EXPORT_SYMBOL(sock_update_classid);
 #endif
 
 /**